src/beamlime/applications/handlers.py

```diff
@@ -125,25 +136,42 @@ def handle(self, message: DataPieceReceived) -> WorkflowResultUpdate | None:
         else:
             self.info("Ignoring data for %s", name)
             return
-        npulse = 100
-        if self._pulse % npulse == 0:
+        if self._pulse % self.n_pulses == 0:
             results = {}
             for name, det in self._detectors.items():
                 buffer = self._buffer[name]
                 if len(buffer):
                     det.add_counts(np.concatenate(buffer))
                     buffer.clear()
                 for window in (1,):
-                    key = f'{name.split("/")[-1]} window={window*npulse}'
+                    key = f'{name.split("/")[-1]} window={window*self.n_pulses}'
```

Comment:
Can you undo all changes to `__init__` and `handle`? It will run into big conflicts with what I am doing.